modules/programs/hexchat.nix

```diff
@@ -0,0 +1,256 @@
+              default = [ ];
+            };
+            autojoin = mkOption {
+              type = nullOr (listOf str);
+              description =
+                "Channels list to autojoin on connecting to server.";
+              example = ''[ "##linux" "#nix" ]'';
+              default = [ ];
+            };
+            commands = mkOption {
+              type = nullOr (listOf str);
+              description = "Commands to be executed on connecting to server.";
+              example = ''[ "ECHO Greetings fellow Nixer! ]'';
+              default = [ ];
+            };
+            login_method = mkOption {
+              type = with lib.types; nullOr int;
+              description = ''
+                null => Default
+                                1 => NickServ (/MSG NickServ + password)
+                                2 => NickServ (/NICKSERV + password)
+                                4 => Challenge Auth (username + password)
+                                6 => SASL (username + password)
+                                7 => Server password (/PASS password)
+                                10 => SASL EXTERNAL (cert)
+                                9 => Custom... (connect commands)
+                              '';
+              default = null;
+            };
+          };
+        };
+      in mkOption {
+        type = nullOr (attrsOf modChannelOption);
+        default = null;
+        description = "Configures '~/.config/hexchat/servlist.conf'";
+        example = ''
+          {
+                        freenode = {
+                          autojoin = [
+                            "#home-manager"
+                            "##linux"
+                            "#nixos"
+                          ];
+                          charset = "UTF-8 (Unicode)";
+                          commands = [
+                            "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+                          ];
+                          login_method = 6;
+                          nickname = "my_nickname";
+                          nickname2 = "my_secondchoice";
+                          options = {
+                            accept_invalid_ssl_certificates = false;
+                            autoconnect = true;
+                            bypass_proxy = true;
+                            connect_to_selected_server_only = true;
+                            use_global_user_information = false;
+                            use_ssl_for_all_servers = false;
+                          };
+                          password = "my_password";
+                          real_name = "my_realname";
+                          servers = [
+                            "chat.freenode.net"
+                            "irc.freenode.net"
+                          ];
+                          user_name = "my_username";
+                        };
+                      }'';
+      };
+    conf = mkOption {
```

Comment:
I think this is traditionally named as `extraConfig`, or if following RFC42, `settings`.

Reply:
I have renamed field name from "conf" to "settings". Is there something else you find relevant? Is it fine now?

Reply:
This is not really following RFC42 (where you would use freeform settings), so maybe `extraConfig` would be better in this case.

But you may also leave this decision to maintainers, for me LGTM.

Reply:
Considering RFC42 settings, the pkgs.formats that mostly resembles HexChat config is INI, but it is still different, HexChat add spaces to equal sign (" = "), and has no sections. The way it is now exactly mimics how HexChat expects it. [I have not tested if it would work with INI. I have checked file contents instead and noticed this difference.]

`extraConfig` is comprised of unstructured strings. Which is not the case with current settings. While current implementation is not using RFC42 proposed implementation, the interface is equal. Current implementation is structured, allows to define values in isolation, merge them, etc. I don't see any meaningful difference to RFC42.

Thoughts?

Reply:
You [can use](https://github.com/NixOS/nixpkgs/blob/master/lib/generators.nix#L80-L83) `lib.generators.toKeyValue { mkKeyValue = lib.generators.mkKeyValueDefault {} " = "; }` probably. Probably RFC42 has some implementation of it using freeform but I don't know exactly where it is implemented.

Reply:
> extraConfig is comprised of unstructured strings.

Not necessary, it can be a merge of structured settings too: https://github.com/nix-community/home-manager/blob/master/modules/programs/rofi.nix#L374-L385. RFC42 is most specifically about the usage of freeform and some guidelines (for example, non exhaustive settings), that is clearly is not the case here.

Reply:
> Probably RFC42 has some implementation of it using freeform but I don't know exactly where it is implemented.

BTW, you don't necessary need to use the generator from RFC42, you can still use `lib.generators.toKeyValue` or your own generator for it. The most important part of RFC42 is really the freeform settings usage.